runtime/service.go

```diff
@@ -67,45 +65,43 @@ func init() {
 
 const (
 	defaultVsockPort     = 10789
+	minVsockIOPort       = uint32(11000)
 	supportedMountFSType = "ext4"
 
 	vmIDEnvVarKey = "FIRECRACKER_VM_ID"
 
-	varRunDir                  = "/var/run/firecracker-containerd"
-	firecrackerSockName        = "firecracker.sock"
-	firecrackerLogFifoName     = "fc-logs.fifo"
-	firecrackerMetricsFifoName = "fc-metrics.fifo"
-
-	// TODO these strings are hardcoded throughout the containerd codebase, it may
-	// be worth sending them a PR to define them as constants accessible to shim
-	// implementations like our own
-	shimAddrFileName = "address"
-	shimLogFifoName  = "log"
-	ociConfigName    = "config.json"
+	varRunDir = "/var/run/firecracker-containerd"
+)
+
+var (
+	// type assertions
+	_ taskAPI.TaskService = &service{}
+	_ shim.Init           = NewService
+
+	sysCall = syscall.Syscall
 )
 
 // implements shimapi
 type service struct {
+	vm.TaskManager
```

Comment:
I would much prefer if this was not embedded as it can make debugging more difficult.

Reply:
Happy to change this, but I'm curious what about embedded fields makes debugging more difficult?

Reply:
Generally when you go searching for a method or field, it isn't on the object but the embedded one, making it not as obvious when viewing the code, and if you start embedding multiple structs it becomes even crazier.